src/Symfony/Component/HttpKernel/EventListener/ValidationFailedExceptionListener.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\ExceptionEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Serializer\Exception\NotEncodableValueException;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Validator\Exception\ValidationFailedException;
+
+/**
+ * @author Konstantin Myakshin <molodchick@gmail.com>
+ */
+final class ValidationFailedExceptionListener implements EventSubscriberInterface
+{
+    public function __construct(private readonly SerializerInterface $serializer)
+    {
+    }
+
+    public function onKernelException(ExceptionEvent $event): void
+    {
+        $throwable = $event->getThrowable();
+        if (!$throwable instanceof ValidationFailedException) {
+            return;
+        }
+
+        $request = $event->getRequest();
+        $format = $request->getPreferredFormat('json');
+
+        try {
+            $data = $this->serializer->serialize($throwable->getViolations(), $format);
+        } catch (NotEncodableValueException) {
+            return;
+        }
+
+        $response = new Response(
+            $data,
+            Response::HTTP_UNPROCESSABLE_ENTITY,
```

Comment:
this should be a 404 when the failure comes from an unprocessable query string (422 is for the body of the request)
can you double check this please?

Reply:
why 404 for invalid query string? Looks strange, 400/422 semantically fits better IMHO.

Reply:
ChatGPT :magic_wand: 

![ChatGPT-400_Bad_Request_for_Query_Validation](https://user-images.githubusercontent.com/191082/226402758-6329c2e4-2f7e-4f01-9dd4-5480060c545b.png)

Reply:
See discussion about this in #49134. 422 is definitely wrong, and 404 is just appropriate. ChatGPT is stupid, we are the ones with a brain.

Reply:
chatgpt is not stupid. http is confusing :D

Reply:
@nicolas-grekas done